### DISCUSSION TITLE
Automatic Contract Verification on Sepolia Network

### DISCUSSION THREAD

**Discussion by tcrin** [Cyfrin/foundry-full-course-cu#3015] [Q&A]
Hi everyone,

In the course, when deploying a contract to the Sepolia network, we are supposed to verify the contract manually. However, every time I deploy a contract, it gets verified automatically. I’m not sure why this is happening.

❓Could anyone help explain why this is occurring?
If there’s any documentation or resources related to this process, it would be greatly appreciated!

![Screenshot from 2024-12-09 22-18-00](https://github.com/user-attachments/assets/afaa5db6-9064-4d2a-818d-5d47978d64a7)

**Comment by EngrPips** [comment]
It is getting verified because you are using the `--verify` flag in your deployment command, and you are providing your etherscan api key, which authenticates the verification. That being said are you not pleased that you were able to verify your contract on deployment?

**Reply by tcrin** [reply]
Oh no, I really like this method. It's just that I’m curious because in the lesson, the instructor also deployed the script with the --verify flag, but still had to verify it manually. So I was wondering

**Reply by EngrPips** [reply]
Sometimes, you may use the `--verify` flag, but for one reason or another the contract might not get verified, and at that point, you need to verify the contract manually. I like it when everything gets verified at deployment though

**Reply by LordAdhaar** [reply]
True
